enhancements/olm/bundle-version-overrides.md

```diff
@@ -0,0 +1,486 @@
+3. Bundles with `release` values ordered before those without
+
+**Explicit Release Specification:**
+
+An operator author builds a bundle with explicit release information:
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: my-operator-v1.0.0-0.20250124000000
+spec:
+  version: 1.0.0
+  release: 0.20250124000000
+  # ... rest of CSV
+```
+
+This bundle, when rendered by `opm`, produces:
+```yaml
+schema: olm.bundle
+name: amq-broker-operator.v
+type: olm.bundle
+...
+properties:
+   - type: olm.package
+     value:
+       packageName: my-operator
+       version: 1.0.0
+       release: 0.20250124000000
+...
+```
+
+**Backward Compatibility:**
+
+Bundles with `olm.substitutesFor` annotation and semver build metadata (e.g., `7.10.2-opr-1+0.1747217191.p`) automatically extract release information from build metadata.
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+...
+spec:
+...
+  version: 7.12.4-opr-1+0.1747217191.p
+...
+```
+
+**Proposed State (After):**
+
+The same bundle now populates both fields:
+```yaml
+schema: olm.bundle
+name: amq-broker-rhel8.v7.10.2-opr-1-0.1747217191.p
+type: olm.package
+value:
+  packageName: amq-broker-rhel8
+  version: 7.10.2-opr-1
+  release: 0.1747217191.p
+```
+
+Build metadata (everything after `+`) is extracted as release version when `olm.substitutesFor` annotation is present.
+
+**Bundle Ordering Priority (Future):**
+
+Priority rules for multiple bundles:
+1. Compare versions first
+2. If equal, compare `release` values (higher wins)
+3. Prefer bundles with `release` over those without
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The proposed change should have no specific impact to HCP. 
+
+#### Standalone Clusters
+
+The proposed change should have no specific impact to standalone clusters. 
+
+#### Single-node Deployments or MicroShift
+
+The proposed change should have no specific impact to SNO/MicroShift clusters. 
+
+### API Extensions
+
+Adds optional `release` field to bundle/catalog schemas:  CSV in operator-framework/api, and `olm.bundle` in operator-framework/operator-registry. Pure additive change—no CRDs, webhooks, or finalizers modified. Change confined to bundle metadata schema in FBC catalogs and operator-registry databases.
+
+### Implementation Details/Notes/Constraints
+
+**Schema Changes:**
+
+The package schema in operator-registry (FBC declcfg) is extended to include an optional `release` field:
+
+```go
+type Package struct {
+    PackageName string `json:"packageName"`
+    Version     string `json:"version"`
+    Release     string `json:"release,omitzero"` // New optional field
+}
+```
+
+**Extraction Logic:**
+
+Hierarchical extraction strategy ([operator-registry PR #1792](https://github.com/operator-framework/operator-registry/pull/1792)):
+
+1. Primary: Extract from CSV `spec.release` field
+2. Fallback: Parse from version build metadata (portion after `+`)
```

Comment:
Do we fallback to extracting from build metadata only if `olm.substitutesFor` is present?